diff --git a/python/sglang/srt/configs/model_config.py b/python/sglang/srt/configs/model_config.py
index 1c31cba3f1..f6ed32e4f2 100644
--- a/python/sglang/srt/configs/model_config.py
+++ b/python/sglang/srt/configs/model_config.py
@@ -372,7 +372,7 @@ class ModelConfig:
         rope_scaling = getattr(self.hf_text_config, "rope_parameters", None) or getattr(
             self.hf_text_config, "rope_scaling", {}
         )
-        self.is_lm_only = getattr(self.hf_config, "language_model_only", False)
+        self.is_lm_only = language_model_only or getattr(self.hf_config, "language_model_only", False)
         self.model_is_mrope = (
             not self.is_lm_only
             and rope_scaling is not None
diff --git a/python/sglang/srt/layers/rotary_embedding/mrope_rope_index.py b/python/sglang/srt/layers/rotary_embedding/mrope_rope_index.py
index 62df65330d..b00999478c 100644
--- a/python/sglang/srt/layers/rotary_embedding/mrope_rope_index.py
+++ b/python/sglang/srt/layers/rotary_embedding/mrope_rope_index.py
@@ -57,6 +57,8 @@ def get_rope_index(
     second_per_grid_ts: Optional[torch.Tensor] = None,
     **kwargs,
 ) -> Tuple[torch.Tensor, torch.Tensor]:
+    if model_type == "qwen3_8_flash_next":
+        model_type = "qwen4_exp"
     if model_type == "qwen3_omni_moe":
         return get_rope_index_qwen3_omni(
             spatial_merge_size,
diff --git a/python/sglang/srt/server_args.py b/python/sglang/srt/server_args.py
index 8af26df97f..c62214efa3 100644
--- a/python/sglang/srt/server_args.py
+++ b/python/sglang/srt/server_args.py
@@ -7749,7 +7749,10 @@ class ServerArgs:
         except Exception:
             return False
 
-    LANGUAGE_MODEL_ONLY_ARCHITECTURES = ("MuseGlimmerForConditionalGeneration",)
+    LANGUAGE_MODEL_ONLY_ARCHITECTURES = (
+        "MuseGlimmerForConditionalGeneration",
+        "Qwen4ExpForConditionalGeneration",
+    )
 
     def _handle_language_model_only(self):
         if not self.language_model_only:
